Tangem/UIComponents/CardsInfoPagerView/CardsInfoPagerView.swift

```diff
@@ -168,13 +191,25 @@ struct CardsInfoPagerView<
                 pageSwitchProgress = abs(value.translation.width / proxy.size.width)
             }
             .onEnded { value in
+                let totalWidth = proxy.size.width
+
+                // Predicted translation takes the gesture's speed into account,
+                // which makes page switching feel more natural.
+                let predictedTranslation = value.predictedEndLocation.x - value.startLocation.x
                 let newIndex = nextIndexToSelectClamped(
-                    translation: value.translation.width,
-                    totalWidth: proxy.size.width,
+                    translation: predictedTranslation,
+                    totalWidth: totalWidth,
                     nextPageThreshold: pageSwitchThreshold
                 )
-                pageSwitchProgress = newIndex == selectedIndex ? 0.0 : 1.0
-                selectedIndex = newIndex
+
+                cumulativeHorizontalTranslation += value.translation.width
+                previouslySelectedIndex = selectedIndex
+
+                withAnimation(pageSwitchAnimation) {
+                    cumulativeHorizontalTranslation = -CGFloat(newIndex) * totalWidth
+                    pageSwitchProgress = newIndex == selectedIndex ? 0.0 : 1.0
```

Comment:
NP: сравнения в тернарных операторах лучше оборачивать в скобки, удобнее читать

```suggestion
                    pageSwitchProgress = (newIndex == selectedIndex) ? 0.0 : 1.0
```

Reply:
Поправил, этот пункт кандидат на внесение в кодстайл)